airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -501,83 +561,41 @@ def create_pod_request_obj(self) -> k8s.V1Pod:
         if self.do_xcom_push:
             self.log.debug("Adding xcom sidecar to task %s", self.task_id)
             pod = xcom_sidecar.add_xcom_sidecar(pod)
-        return pod
 
-    def create_new_pod_for_operator(self, labels, launcher) -> Tuple[State, k8s.V1Pod, Optional[str]]:
-        """
-        Creates a new pod and monitors for duration of task
-
-        :param labels: labels used to track pod
-        :param launcher: pod launcher that will manage launching and monitoring pods
-        :return:
-        """
-        self.log.debug(
-            "Adding KubernetesPodOperator labels to pod before launch for task %s", self.task_id
-        )
+        labels = self._create_labels_for_pod(context)
+        self.log.info("creating pod with labels %s and launcher %s", labels, self.launcher)
 
         # Merge Pod Identifying labels with labels passed to operator
-        self.pod.metadata.labels.update(labels)
+        self.log.debug("Adding KubernetesPodOperator labels to pod before launch for task %s", self.task_id)
```

Comment:
```suggestion
```

Not sure this is really that useful? We log the labels themselves as INFO above.